generated/generics/generate.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+if [ -z $GOPATH ]; then
+  echo 'GOPATH is not set'
+  exit 1
+fi
+
+GENERIC_MAP_PATH=${GOPATH}/src/github.com/m3db/m3ninx/vendor/github.com/m3db/m3x/generics/hashmap
+GENERIC_MAP_IMPL=${GENERIC_MAP_PATH}/map.go
+
+if [ ! -f "$GENERIC_MAP_IMPL" ]; then
+  echo "${GENERIC_MAP_IMPL} does not exist"
+  exit 1
+fi
+
+GENERATED_PATH=${GOPATH}/src/github.com/m3db/m3ninx/index/segment/mem
```

Comment:
It seems this is pretty specific to the current packages we're generating, is it worth having perhaps just a short README to explain how to add another package in the future?

Reply:
Sure, I put down a few paragraphs in the script file.